src/files/BrsFile.ts

```diff
@@ -537,6 +537,20 @@ export class BrsFile {
                 });
             }
 
+            //add every loop-defined variable
+            func.body.walk(createVisitor({
```

Comment:
Mixed feeling about this walk: 
- how often is it executed?
- it should probably be collected in the references,
- but on the other hand, doing a late walking only for the function you're interested in can be good.

Reply:
As for the assignment references, they really only make sense for the function you need completion in, so it should be done on demand.

Reply:
I don't see any benefits to converting this data to on-demand because `Scope.diagnosticDetectShadowedLocalVars()` references `FunctionScope.variableDeclarations`, which is run on every validation cycle. 

However, we definitely could move these into the references. I'll give that a try and see how it goes.

Reply:
Based on my refactor, the `FunctionScope` concept has been eliminated completely so this is no longer relevant.